subprojects/core/src/main/java/org/gradle/internal/scan/config/BuildScanConfigManager.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.scan.config;
+
+import org.gradle.BuildAdapter;
+import org.gradle.StartParameter;
+import org.gradle.api.invocation.Gradle;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.internal.event.ListenerManager;
+
+import java.util.Map;
+
+/**
+ * This is the meeting point between Gradle and the build scan plugin during initialization.
+ * This is effectively build scoped.
+ */
+class BuildScanConfigManager implements BuildScanConfigInit, BuildScanConfigProvider {
+
+    private static final Logger LOGGER = Logging.getLogger(BuildScanConfigManager.class);
+
+    private static final String HELP_LINK = "https://gradle.com/scans/help/gradle-cli";
+    private static final String SYSPROP_KEY = "scan";
+
+    private final StartParameter startParameter;
+    private final ListenerManager listenerManager;
+    private final BuildScanPluginCompatibilityEnforcer compatibilityEnforcer;
+
+    private State state = State.DEFAULTED;
+    private boolean collected;
+
+    BuildScanConfigManager(StartParameter startParameter, ListenerManager listenerManager, BuildScanPluginCompatibilityEnforcer compatibilityEnforcer) {
+        this.startParameter = startParameter;
+        this.listenerManager = listenerManager;
+        this.compatibilityEnforcer = compatibilityEnforcer;
+    }
+
+    @Override
+    public void init() {
+
+        boolean disabledViaSysProp = false;
+        boolean enabledViaSysProp = false;
+        Map<String, String> sysProps = startParameter.getSystemPropertiesArgs();
+        if (sysProps.containsKey(SYSPROP_KEY)) {
+            String sysProp = sysProps.get(SYSPROP_KEY);
+            if ("false".equals(sysProp)) {
+                disabledViaSysProp = true;
+            } else {
+                enabledViaSysProp = true;
+            }
```

Comment:
Not sure whether we should handle `-Dscan=foo" differently -> right now it is treated as true.